### PR TITLE
app-backup/b2-2.1.0.ebuild BackBlaze's b2 CLI, with dependencies.

### PR DESCRIPTION
**Pull request by ezzieyguywuf**
This is my first pull request to Gentoo, please let me know what I need to do or change in order to get this merged.

### CODE REVIEW

**Comment by ezzieyguywuf** [comment]
Fixed QA issues, and added GCO signoff.

**Comment by ezzieyguywuf** [comment]
Added the required proxy information to the metadata.xml file

File: app-backup/b2/b2-2.1.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{5..8} )
```
Comment (thesamesam) [inline app-backup/b2/b2-2.1.0.ebuild:6]: 5? Get outta town 😃 

Do 7,8,9 if you can.

File: app-backup/b2/b2-2.1.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+inherit distutils-r1
+
+DESCRIPTION="The command-line tool for BackBlaze's B2 product."
+HOMEPAGE="https://github.com/Backblaze/B2_Command_Line_Tool"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
```
Comment (thesamesam) [inline app-backup/b2/b2-2.1.0.ebuild:20]: Only needed if you have test dependencies. Also, distutils_enable_tests does this for you!

File: dev-python/b2sdk/b2sdk-1.2.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+inherit distutils-r1
+
+DESCRIPTION="The client library for BackBlaze's B2 product"
+HOMEPAGE="https://github.com/Backblaze/b2-sdk-python"
+SRC_URI="https://github.com/Backblaze/b2-sdk-python/releases/download/v${PV}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
```
Comment (thesamesam) [inline dev-python/b2sdk/b2sdk-1.2.0.ebuild:16]: Unnecessary, d_e_t does this for us

File: dev-python/b2sdk/b2sdk-1.2.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+PATCHES=(
+	"${FILESDIR}/${P}_skip-integration-test.patch"
+	"${FILESDIR}/${P}_skip-license-header-test.patch"
```
Comment (thesamesam) [inline dev-python/b2sdk/b2sdk-1.2.0.ebuild:20]: - ${P}_ looks odd, usually we do ${P}-...
- I usually do ${P} if it's an upstreamed patch or ${PN}-versionItIsBasedOn if it's not. This is just a style/hint thing though.

File: dev-python/b2sdk/b2sdk-1.2.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+		>=dev-python/requests-2.9.1[${PYTHON_USEDEP}]
+		>=dev-python/tqdm-4.5.0[${PYTHON_USEDEP}]
+	')
+"
+
+distutils_enable_tests pytest
+
+BDEPEND+=" test? (
+	$(python_gen_cond_dep '
+		>=dev-python/pytest-6.1.1[${PYTHON_USEDEP}]
```
Comment (thesamesam) [inline dev-python/b2sdk/b2sdk-1.2.0.ebuild:36]: d_e_t does pytest for you here at least

File: dev-python/phx-class-registry/phx-class-registry-3.0.5.ebuild [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+inherit distutils-r1
+
+DESCRIPTION="An intersection of the Registry and Factory pattern"
+HOMEPAGE="https://github.com/todofixthis/class-registry"
+SRC_URI="https://github.com/todofixthis/class-registry/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
```
Comment (thesamesam) [inline dev-python/phx-class-registry/phx-class-registry-3.0.5.ebuild:16]: Unnecessary, d_e_t does the IUSE and RESTRICT for us (also adds test? ( ${RDEPEND} )).

File: dev-python/phx-class-registry/phx-class-registry-3.0.5.ebuild [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{5..8} pypy3 )
```
Comment (thesamesam) [inline dev-python/phx-class-registry/phx-class-registry-3.0.5.ebuild:6]: No Python 3.5! Test with 7,8,9 (and maybe 6) if you can.

File: dev-python/b2sdk/files/b2sdk-1.2.0_skip-license-header-test.patch [hunk 8]
```diff
@@ -0,0 +1,12 @@
+diff --git a/test/static/test_licenses.py b/test/static/test_licenses.py
+index babda33..b682f6d 100644
+--- a/test/static/test_licenses.py
++++ b/test/static/test_licenses.py
+@@ -14,6 +14,7 @@ from itertools import islice
+ import pytest
+ 
+ 
++@pytest.mark.skip(reasot="This is an upstream issue, not gentoo")
```
Comment (thesamesam) [inline dev-python/b2sdk/files/b2sdk-1.2.0_skip-license-header-test.patch:9]: - reasot? :)
- Got a link? Or do you mean that it's like a linter => the result doesn't matter to us?

File: app-backup/b2/b2-2.1.0.ebuild (hunk 1 above)
Comment (ezzieyguywuf) [reply app-backup/b2/b2-2.1.0.ebuild:6]: I copied the supported versions that were listed on pypi.

Is it preferable to use a gentoo-specific list? The max version listed was 3.8, wants the criteria for including higher versions in am ebuild?

File: dev-python/b2sdk/b2sdk-1.2.0.ebuild (hunk 4 above)
Comment (ezzieyguywuf) [reply dev-python/b2sdk/b2sdk-1.2.0.ebuild:20]: Good to know I'll make this change (on my other ebuild submission too)

File: dev-python/b2sdk/b2sdk-1.2.0.ebuild (hunk 5 above)
Comment (ezzieyguywuf) [reply dev-python/b2sdk/b2sdk-1.2.0.ebuild:36]: I think I had some issues but likely due to putting the d_e_t call in the wrong place. I'll fix this.

File: dev-python/b2sdk/files/b2sdk-1.2.0_skip-license-header-test.patch (hunk 8 above)
Comment (ezzieyguywuf) [reply dev-python/b2sdk/files/b2sdk-1.2.0_skip-license-header-test.patch:9]: hrm, I wasn't running this in a "clean" environment, now that I am I am seeing different test errors, even disabling this patch. I'll dig into this.

Comment (ezzieyguywuf) [reply dev-python/b2sdk/files/b2sdk-1.2.0_skip-license-header-test.patch]: Turns out this was a red herring, re-enabled the test and everything worked fine.

**Comment by ezzieyguywuf** [comment]
This can be closed, as this has been made available on guru with commit [1d927](https://github.com/gentoo/guru/commit/1d9279bbe078badae5c434759b070022ae23147c)

**Pull request closed without merging**